### PR TITLE
Adding support for Snakemake

### PR DESCRIPTION
**Pull request by dfm**
This PR adds support for [Snakemake](https://snakemake.readthedocs.io) files which, for our purposes, should have the same syntax as Python. I'm definitely not a Perl expert, but I've copied the Python behavior and added a test. Please let me know if there's anything else that I should check. Thanks!!

### CODE REVIEW

File: cloc [hunk 1]
```diff
@@ -10338,6 +10341,14 @@ sub set_constants {                          # {{{1
                                 [ 'smarty_to_C'                    ],
                                 [ 'call_regexp_common'  , 'C'      ],
                             ],
+    'Snakemake'          => [
+                                [ 'remove_matches'      , '/\*'    ],
```
Comment (AlDanial) [inline cloc:10345]: I'm guessing the `remove_matches` using `'/\*'` and `'\*/'` were intended to capture C style comments?  That's not what will happen, instead these will strip only from these markers to the end of the line.  Just remove these two lines (10345 & 10346) and you'll be fine.   `/*` .. `*/` comments are captured with `call_regexp_common` `C`.

File: cloc [hunk 2]
```diff
@@ -11263,6 +11275,7 @@ sub set_constants {                          # {{{1
     'PHP'                          => 3.50,
     'Jupyter Notebook'             => 4.20,
     'Python'                       => 4.20,
+    'Snakemake'                    => 4.20, # This is a guess. Copied from Python, because Snakemake supports Python syntax
```
Comment (AlDanial) [inline cloc:11278]: The value is fine.  You can remove the comment.

**Comment by dfm** [comment]
Thanks for the comments @AlDanial! I've addressed them both.

**Comment by AlDanial** [comment]
Thanks for the contribution.

**Comment by AlDanial** [comment]
Ooops!  Meant to merge... hit the wrong button.

**Comment by YuanfengZhang** [comment]
Sorry but why no Snakemake support now? (Ubuntu 24.04 apt-get source, v1.90)

**Comment by AlDanial** [comment]
Snakemake exists in cloc v2.00 and higher (the current release is v2.04).  The Ubuntu package is old.  Get 2.04 from the Releases area, https://github.com/AlDanial/cloc/releases

**Pull request merged**